### PR TITLE
support of multiple clusters in test-cluster.sh

### PR DESCRIPTION
**Pull request by hrdlotom**
dynamic cluster management in test-cluster.sh (create, add, restart, stop, clean)

Before testing for the first time remove zk, waltz storage and waltz server docker images

try:
./bin/test-cluster.sh start 3                  #creates 3 clusters, each consists of one server and storage node
./bin/test-cluster.sh restartCluster 1   #restarts cluster 1
./bin/test-cluster.sh stopCluster 0      #stops cluster 0
./bin/test-cluster.sh addCluster.          #adds new cluster (will be named as cluster 3)
./bin/test-cluster.sh stop                      #stops all running waltz clusters
./bin/test-cluster.sh clean                    #removes all waltz clusters

### CODE REVIEW

File: bin/docker/waltz-server.sh [hunk 1]
```diff
@@ -5,19 +5,25 @@ cmd=$1
...
 imageSource=waltz-server:distDocker
 imageName=com.wepay.waltz/waltz-server
-containerName=waltz-server
+containerName="$2_server"
 networkName=waltz-network
-ports=55180-55182:55180-55182
+configFolder="$2"
+
+if [ $cmd = "start" ]; then
+    portsOccupiedLowerBound=$3
```
Comment (smithakoduri) [inline bin/docker/waltz-server.sh:13]: nit: May be "portsLowerBound" instead of "portsOccupiedLowerBound" ? Similarly, at other places too.

**Review by smithakoduri — COMMENTED** [review]
One small nit..rest LGTM

File: bin/test-cluster.sh [hunk 2]
```diff
@@ -2,41 +2,131 @@
...
+    $DIR/docker/cluster-config-files.sh "$clusterName" $serverPortLowerBound $storagePortLowerBound
+
+    $DIR/docker/cluster.sh create "$clusterName"
+    $DIR/docker/waltz-storage.sh start "$clusterName" $storagePortLowerBound
+    $DIR/docker/add-storage.sh "$clusterName" $storagePortLowerBound
+    $DIR/docker/waltz-server.sh start "$clusterName" $serverPortLowerBound
+}
+
+stopCluster() {
+    $DIR/docker/waltz-server.sh stop "$1"
```
Comment (zzlbuaa) [inline bin/test-cluster.sh:55]: Just had a question regarding stop and clean, in `waltz-server.sh` and also in storage, it is and was not specifically checking the passed-in `$cmd` with "stop" nor use it to do anything, just curious how the stop/clean was handled? What does those scripts do for a stop/clean? Not sure if I'm missing anything.

File: README.md [hunk 3]
```diff
@@ -97,17 +99,17 @@ If the Docker images are not built yet, this script builds them.
...
 
-### Stopping the test cluster
+### Stopping waltz test cluster
 
-    bin/test-cluster.sh stop
+    bin/test-cluster.sh stop <- stop all created clusters
+    bin/test-cluster.sh stop <cluster_name> <- stop cluster with the given cluster name
 
-This stops all three containers. You can resume the cluster using `start` command. All data in zookeeper and storages are preserved.
+This stops waltz containers. You can resume the cluster using `start <cluster_name>` command. All data in zookeeper and storages are preserved.
```
Comment (zzlbuaa) [inline README.md:108]: nit: maybe we can add the descriptions about test-cluster.sh restart in readme, looks it was missed here.

**Review by zzlbuaa — APPROVED** [review]
Overall LGTM!

File: bin/test-cluster.sh (hunk 2 above)
Comment (hrdlotom) [reply bin/test-cluster.sh:55]: the commands are triggered in container.sh. Basically command `source $DIR/container.sh` in `waltz-server.sh` extends `waltz-server.sh` with `container.sh`.

**Pull request merged**